Source/loadsave.cpp

```diff
@@ -2014,12 +2063,25 @@ void SaveGameData()
 			file.WriteBE<int32_t>(monsterId);
 		for (int i = 0; i < ActiveMonsterCount; i++)
 			SaveMonster(&file, Monsters[ActiveMonsters[i]]);
-		for (int missileId : ActiveMissiles)
-			file.WriteLE<int8_t>(missileId);
-		for (int missileId : AvailableMissiles)
-			file.WriteLE<int8_t>(missileId);
-		for (int i = 0; i < ActiveMissileCount; i++)
-			SaveMissile(&file, Missiles[ActiveMissiles[i]]);
+		// Write ActiveMissiles
+		for (size_t activeMissile = 0; activeMissile < std::min(MaxMissilesForSaveGame, Missiles.size()); activeMissile++) {
+			file.WriteLE<int8_t>(activeMissile);
+		}
+		if (MaxMissilesForSaveGame > Missiles.size())
+			file.Skip<int8_t>(MaxMissilesForSaveGame - Missiles.size());
+		// Write AvailableMissiles
+		for (size_t avaiableMissile = Missiles.size(); avaiableMissile < MaxMissilesForSaveGame; avaiableMissile++) {
+			file.WriteLE<int8_t>(avaiableMissile);
+		}
+		file.Skip<int8_t>((MaxMissilesForSaveGame > Missiles.size()) ? Missiles.size() : MaxMissilesForSaveGame);
+		// Write Missile Data
+		size_t wroteMissiles = 0;
+		for (auto &missile : Missiles) {
+			SaveMissile(&file, missile);
+			wroteMissiles += 1;
+			if (wroteMissiles == MaxMissilesForSaveGame)
+				break;
+		}

```

Comment:
This could use `std::for_each_n` maybe?
```suggestion
		std::for_each_n(Missiles.begin(), std::min(Missiles.size(), MaxMissilesForSaveGame), [&file] {
			SaveMissile(&file, missile);
		})
```

Reply:
We will see if the old gcc can handle it 😁

Reply:
Sadly not 😢 
> /root/repo/Source/loadsave.cpp:2072:3: error: 'for_each_n' is not a member of 'std'

Changed back.

Reply:
Thanks for testing :)